homeassistant/components/google_drive/backup.py

```diff
@@ -0,0 +1,273 @@
+"""Backup platform for the Google Drive integration."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncIterator, Callable, Coroutine
+import json
+from typing import Any, Self
+
+from aiohttp import ClientError, MultipartWriter, StreamReader
+
+from homeassistant.components.backup import AgentBackup, BackupAgent, BackupAgentError
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from . import DATA_BACKUP_AGENT_LISTENERS, GoogleDriveConfigEntry
+from .api import AsyncConfigEntryAuth
+from .const import DOMAIN, DRIVE_API_FILES, DRIVE_API_UPLOAD_FILES
+
+# Google Drive only supports string key value pairs as properties.
+# We convert any other fields to JSON strings.
+_AGENT_BACKUP_SIMPLE_FIELDS = [
+    "backup_id",
+    "date",
+    "database_included",
+    "homeassistant_included",
+    "homeassistant_version",
+    "name",
+    "protected",
+    "size",
+]
+
+
+def _convert_agent_backup_to_properties(backup: AgentBackup) -> dict[str, str]:
+    return {
+        k: v if k in _AGENT_BACKUP_SIMPLE_FIELDS else json.dumps(v)
```

Comment:
This feels a little fragile, should we include a `"version": "1"` as a key/value so in the future when fields change in some way, we can handle that (ie `json.loads` below would fail if it's not valid JSON)